### PR TITLE
Update cookbook to match what Commvault did upstream

### PR DESCRIPTION
**Pull request by mvangoor**
Commvault fixed a bug in 11.21 that was not apparent untill 11.24 and then we hit new issues with the API.

The current code is a result of all the updates needed to make it work post 11.21 AND add new features to allow "plan rules"

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,14 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/) and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## 1.0.0
+
+- replace File.open with File.write
```
Comment (shoekstra) [inline CHANGELOG.md]:
```suggestion
- Replace `File.open` with `File.write`
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,14 @@ All notable changes to this project will be documented in this file.
...
+- Commvault has changed the way it handles `restore only` installations and we therefor need to register the fs agent post instllation now
```
Comment (shoekstra) [inline CHANGELOG.md]:
```suggestion
- Commvault has changed the way it handles `restore only` installations, we now need to register the file system agent post installation
```

File: README.md [hunk 3]
```diff
@@ -70,12 +70,13 @@ end
 | use_cache                | [TrueClass, FalseClass] | true                       | By default this cookbook does caching to limit hammering of the API, but with this you can disable it                               |
 | cache_timeout            | Integer                 | 43200                      | The time to live for cache entries before we talk to the API again                                                                  |
 | use_local_login          | [TrueClass, FalseClass] | true                       | By default we use local qlogin with localadmin impersonation, if you would like to add user/pass set this to false                  |
+| plan_name                | [String,nil]            | nil                        | Can be used to assign a plan if at any point a client is switched from unmanaged to managed                                         |
```
Comment (shoekstra) [inline README.md]:
```suggestion
| plan_name                | [String, nil]           | nil                        | Can be used to assign a plan if at any point a client is switched from unmanaged to managed                                         |
```

File: libraries/api.rb [hunk 4]
```diff
@@ -80,6 +101,29 @@ def cv_fs_filter(endpoint, cv_token, subclient_name, filters)
...
+      else
+        Chef::Log.debug 'planEntity not set'
+      end
+      props['subClientProperties'][0].key?('planEntity') && props['subClientProperties'][0]['planEntity'].key?('planName')
+    end
+
+    def cv_fs_subclient_assign_plan(endpoint, cv_token, subclient_name, plan_name)
+      unless subclient_name == 'default'
+        raise 'We are unable to manage any subclient other than default for now'
+      end
```
Comment (shoekstra) [inline libraries/api.rb]:
```suggestion
      raise 'We are unable to manage any subclient other than default for now' unless subclient_name == 'default'

```

Just a preference thing, but if it's short enough for one line it's preferred, and to leave an extra line to make it clear that you're going to return or raise an exception.

File: resources/install.rb [hunk 5]
```diff
@@ -191,7 +215,7 @@
       end
       unless cvlt_registered?
         puts "CommVault registration unsuccesful after #{new_resource.registration_timeout} seconds, uninstalling"
-        # Call uninstall
+        # Call uninstall - TODO
```
Comment (shoekstra) [inline resources/install.rb]:
```suggestion
        # TODO: Call uninstall
```

This is a common way to show things to be done, some editors even pick it up and show different highlighting.

**Review by shoekstra — CHANGES_REQUESTED** [review]

File: resources/install.rb (hunk 5 above)
Comment (mvangoor) [reply resources/install.rb]: Done

File: libraries/api.rb (hunk 4 above)
Comment (mvangoor) [reply libraries/api.rb]: Done

File: README.md (hunk 3 above)
Comment (mvangoor) [reply README.md]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (mvangoor) [reply CHANGELOG.md]: Done

File: CHANGELOG.md (hunk 1 above)
Comment (mvangoor) [reply CHANGELOG.md]: Done

**Review by shoekstra — APPROVED** [review]

**Pull request merged**